tpl/tplimpl/template.go

```diff
@@ -224,6 +224,7 @@ func (t *templateExec) Execute(templ tpl.Template, wr io.Writer, data interface{
 		defer t.Metrics.MeasureSince(templ.Name(), time.Now())
 	}
 
+	t.d.Log.Debug().Printf("Execute template: %q", templ.Name())
```

Comment:
OK, I didn't notice this the first time I looked at this. I'm not a big fan of permanent debug statements in the first place, but especially here in what I would consider to be somewhat of a hot path. There is a cost of doing that debugging (even if it's turned of), so unless it's really important, I'd rather we delete that line.

Reply:
I'd like to keep this as it's valuable debug info.  We should be able to protect this line with an if-statement if we're concerned about the Printf exec time:

```go
if t.d.Cfg.GetBool("debug") {
```
The config lookups are cached.

Even with debug enabled on the Hugo docs site, this doesn't add much to the build time (if you redirection stdout to a file).

Reply:
I amended the PR as requested.